packages/amplication-client/src/VersionControl/CommitsPage.tsx

```diff
@@ -20,42 +27,81 @@ type Props = AppRouteProps & {
   }>;
 };
 
+const MAX_ITEMS_PER_LOADING = 20;
+
 const CommitsPage: React.FC<Props> = ({ match, moduleClass }) => {
   const commitId = match.params.commit;
   const history = useHistory();
 
   const { currentProject, currentWorkspace } = useContext(AppContext);
+  const [commitsCount, setCommitsCount] = useState(1);
+  const [disableLoadMore, setDisableLoadMore] = useState(false);
+
+  const { commits, commitsError, commitsLoading, refetchCommits } = useCommit(
+    MAX_ITEMS_PER_LOADING
+  );
+
+  const [commitsData, setCommitsData] = useState<Commit[]>([]);
+
+  useEffect(() => {
```

Comment:
also this... I'll explain. the hook is already manage the commits as state, it is not necessary to manage another state. instead you should get the commits from the hook as initial value and updated after click "show more"

Reply:
so you wont need this useEffect